src/finch/symbolic/rewriters.py

```diff
@@ -141,15 +164,12 @@ class Fixpoint:
     def __init__(self, rw: RwCallable):
         self.rw = rw
 
-    def __call__(self, x: Term) -> Term | None:
+    def __call__(self, x: T) -> T | None:
         y = self.rw(x)
-        if y is not None:
-            while y is not None and x != y:
-                x = y
-                y = self.rw(x)
-            return x
-        else:
-            return None
+        while y is not None and x != y:
+            x = y
+            y = self.rw(x)
+        return y
```

Comment:
Can we move this back to the old code here? It's hard to do a PR over many files that makes changes to the logic like this. I think this new form may be wrong, but even if it's not, this was only supposed to be a linting pr.